test/gurka/test_ferry_connections.cc

```diff
@@ -26,15 +27,69 @@ ReclassifyFerryConnectionEdge(std::map<std::string, std::string> const& way_desc
                               {"name", "Cape May-Lewes Ferry"},
                               {"destination:forward", "Cape May"},
                               {"destination:backward", "Lewes"}}},
-                            {"DE", {{"highway", "trunk"}, {"name", ""}}}};
+                            {"DE", {{"highway", "trunk"}, {"name", ""}}},
+                            {"FG", {{"highway", "trunk"}, {"name", ""}}},
+                            {"Gg", way_description},
+                            {"gH", way_description},
+                            {"HI",
+                             {{"motor_vehicle", "yes"},
+                              {"motorcar", "yes"},
+                              {"bicycle", "yes"},
+                              {"foot", "no"},
+                              {"duration", "35"},
+                              {"route", "shuttle_train"},
+                              {"name", "Eurotunnel Shuttle"}}},
+                            {"IJ", {{"highway", "trunk"}, {"name", ""}}}};
 
   const auto layout = gurka::detail::map_to_coordinates(ascii_map, gridsize_metres);
 
   auto map = gurka::buildtiles(layout, ways, {}, {}, "test/data/gurka_reclassify_ferry_connections");
   baldr::GraphReader graph_reader(map.config.get_child("mjolnir"));
 
-  auto edge = std::get<1>(gurka::findEdgeByNodes(graph_reader, layout, "B", "b"));
-  return edge->classification();
+  auto Bb_edge = std::get<1>(gurka::findEdgeByNodes(graph_reader, layout, "B", "b"));
+  auto Gg_edge = std::get<1>(gurka::findEdgeByNodes(graph_reader, layout, "G", "g"));
+  return {Bb_edge->classification(), Gg_edge->classification()};
+}
+
+TEST(Standalone, ShortFerry) {
+  const std::string ascii_map = R"(
+          A--B--C--F--D--E--G
+                   |
+                   N
+    )";
```

Comment:
Adding a test case here to cover the issue discovered. Both `CF` and `FD` are ferry edges, but they are both < 2km. This means that `ShortFerry` should return `false` since the ferry consists of > 1 edge. Thus, we should be reclassifying `BC` and `DE` and we check that in this unit test.

Also made sure that this unit test was failing if we temporarily removed this PR's changes to `ShortFerry`.

Reply:
yeah the intuition here is we shouldnt assume the ferry is short just because it has 1 short edge in it. seems like somewhere along the chunnel another drivable edge intersects it and so makes the first edge short but then of course the whole rest of the chunnel is many km long. basically shortFerry was tricked. it still might be better to completely disable shortFerry, you can think of malicious cases where a short single edge ferry is the only way to get from one place to another but it likely doesnt happen in the wild that often (or maybe ever)

Reply:
You may never know when it appears)